saleor/graphql/order/tests/mutations/test_order_bulk_create.py

```diff
@@ -4053,3 +4057,185 @@ def test_order_bulk_create_shipping_price_zero(
     db_order = Order.objects.last()
     assert db_order.shipping_price_net_amount == 0
     assert db_order.shipping_price_gross_amount == 0
+
+
+def test_order_bulk_create_save_addresses_in_customer_book(
+    staff_api_client,
+    permission_manage_orders,
+    permission_manage_orders_import,
+    permission_manage_users,
+    order_bulk_input,
+    customer_user,
+):
+    # given
+    order = order_bulk_input
+    # change the field in billing address to ensure that both shipping and billing
+    # addresses are the new one
+    order_bulk_input["billingAddress"]["postalCode"] = "00-000"
+    order["saveShippingAddress"] = True
+    order["saveBillingAddress"] = True
+
+    staff_api_client.user.user_permissions.add(
+        permission_manage_orders_import,
+        permission_manage_orders,
+        permission_manage_users,
+    )
+    variables = {
+        "orders": [order],
+        "stockUpdatePolicy": StockUpdatePolicyEnum.SKIP.name,
+    }
+    user_address_count = customer_user.addresses.count()
+
+    # when the bulk order is called with save address options set
+    response = staff_api_client.post_graphql(ORDER_BULK_CREATE, variables)
+    content = get_graphql_content(response)
+
+    # then the addresses are saved in customer address book
+    assert content["data"]["orderBulkCreate"]["count"] == 1
+    data = content["data"]["orderBulkCreate"]["results"]
+    assert not data[0]["errors"]
+
+    order = data[0]["order"]
+    assert order["created"]
+
+    db_order = Order.objects.get()
+    assert db_order.billing_address
+    assert db_order.shipping_address
+    assert db_order.draft_save_billing_address is None
+    assert db_order.draft_save_shipping_address is None
+    assert customer_user.addresses.count() == user_address_count + 2
```

Comment:
Can we also confirm that the `db_order.shipping_address.id != user.addresses[x].id`? (same for billing)